src/algorithms/models/vertex.ts

```diff
@@ -40,17 +40,72 @@ class Vertex {
    * Counted as the number of neighbors.
   */
   get outDegree(): number {
-    return this.neighbors.length;
+    return this.neighbors.size;
   }
 
   /**
    * Add a neighbor with edge
    */
   addNeighbor(costs: EdgeCost, vertex: Vertex): void {
-    this.neighbors.push({
+    this.neighbors.add({
       costs,
       vertex
     });
+    vertex.inDegree += 1;
+  }
+
+  /**
+   * Remove a neighbor of this vertex
+   * Outcoming edge of h=this vertex has to be passed as a parameter. The edge that matches the passed one will be
+   * removed from the neighbors array, therefore neighbor is removed.
+   * @param edge Edge to a neighbor that will be removed
+   */
+  removeNeighbor(edge: {vertex: Vertex, costs: EdgeCost}): void {
+    this.neighbors.delete(edge);
+    edge.vertex.inDegree -= 1;
+  }
+
+  /**
+   * Method verifies whether there is an edge back from the 'from' vertex back to this vertex
+   * If there is at least one edge from 'from' vertex' to this vertex true is returned. Otherwise false.
+   * @param from : Vertex Vertex to get back from
+   * @return boolean
+   */
+  canGetBack(from: Vertex): boolean {
+    for (const edge of from.neighbors) {
+      if (edge.vertex === this) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Method verifies whether the inDegree and outDegree of the vertex is the same and whether it is equal to the degree
+   * True is returned when both degrees are equal to the degree parameter. Otherwise false.
+   * @param degree : number outDegree and inDegree that should be equal to
+   * @return boolean
+   */
+  equalDegrees(degree: number): boolean {
```

Comment:
What's the reason that you pass the compared number as a parameter instead of just returning the degree and making the comparison outside the function? That makes it more complicated in my eyes than it could be.

Reply:
I use it just in cases where the degrees has to be equal and they has to be equal to a concrete value. Instead of another comparision outside function this does it in one place.